### ISSUE TITLE
Abstract method false-positive when using TypeVarTuple

### ISSUE THREAD

**Issue by kdkavanagh** [PyCQA/pylint#10972]
### Bug description
False positive specifically when using TypeVarTuple / PEP-695 generics

```python

from __future__ import annotations

from abc import ABC, abstractmethod
from typing import Generic, TypeVar

T = TypeVar("T")


class Base(Generic[T], ABC):
    @abstractmethod
    def get_item(self) -> T: ...


class Middle[NpDtype, *Shape]:
    """Implements the abstract method from Base."""

    def get_item(self) -> NpDtype:
        raise NotImplementedError


class Concrete[NpDtype, *Shape](Middle[NpDtype, *Shape], Base[NpDtype]):
    """All abstract methods are satisfied via Middle."""


obj = Concrete()  # E0110 false positive
```

### Configuration

```ini

```

### Command used

```shell
pylint -vv
```

### Pylint output

```python
************* Module pylint_e0110_repro                                                             
     pylint_e0110_repro.py:47:6: E0110: Abstract class 'Concrete' with abstract methods instantiated     
     (abstract-class-instantiated)
```

### Expected behavior

No error

### Pylint version

```shell
$ pylint --version
pylint 4.0.5                                                                                             
astroid 4.0.4                                                                                            
Python 3.12.6 (main, Oct  9 2024, 11:19:43) [GCC 13.2.0]
```

### Additional dependencies

```python

```

**Comment by DanielNoord** [comment]
More minimal reproducer is:

```python
from abc import ABC, abstractmethod


class Base(ABC):
    @abstractmethod
    def get_item(self) -> None: ...


class Middle[*Shape]:
    """Implements the abstract method from Base."""

    def get_item(self) -> None:
        raise NotImplementedError


class Concrete[*Shape](Middle[*Shape], Base):
    """All abstract methods are satisfied via Middle."""


obj = Concrete()  # E0110 false positive
```

Changing it to `T` instead of `*Shape` does work, so this is likely an issue with the `*`.

**Comment by ekanshul** [comment]
This no longer reproduces on current `main` (pylint 4.1.0-dev0, astroid 4.2.0b5, Python 3.12.13): running the snippet from the report with the message enabled produces no output. It may be worth closing, or adding the snippet as a regression test.
